doc/mesh-adapter-api.txt

```diff
@@ -0,0 +1,137 @@
+BlueZ D-Bus Mesh Adapter API description
+********************************
+
```

Comment:
Some intro would be nice (what and why, what are the current limitations).
Mention that the adapters are persistent.